include/commissioner/commissioner.hpp

```diff
@@ -173,7 +173,18 @@ class Commissioner
      * @note @p aResponseData is guaranteed to be not null only when @p aError == Error::kNone.
      *       Otherwise, @p aResponseData should never be accessed.
      */
-    template <class T> using Handler = std::function<void(const T *aResponseData, Error aError)>;
+    template <typename T> using Handler = std::function<void(const T *aResponseData, Error aError)>;
+
+    /**
+     * The petition result handler.
+     *
+     * @param[in] aActiveCommissionerId  A active commissioner Id. nullable.
+     * @param[in] aError                 A error code.
+     *
+     * @note There is another active commissioner if @p aError != Error::kNone
+     *       and @p aActiveCommissionerId is not null.
+     */
+    using PetitionHandler = std::function<void(const std::string *aActiveCommissionerId, Error aError)>;
```

Comment:
```suggestion
    using PetitionHandler = std::function<void(const std::string *aExistingCommissionerId, Error aError)>;
```

Reply:
Is `aActiveCommissionerId` only valid on error? If so, I prefer to change to `aExistingCommissionerId`.

Reply:
Yes.